lib/LocalDevManagerV2.ts

```diff
@@ -0,0 +1,577 @@
+  isGithubLinked: boolean;
+  watcher: FSWatcher | null;
+  uploadWarnings: { [key: string]: boolean };
+  components: { [key: string]: IntermediateRepresentationNodeLocalDev };
+  activeApp: AppIRNode | null;
+  activePublicAppData: PublicApp | null;
+  env: Environment;
+  publicAppActiveInstalls: number | null;
+  projectSourceDir: string;
+  mostRecentUploadWarning: string | null;
+
+  constructor(options: LocalDevManagerConstructorOptions) {
+    this.targetProjectAccountId = options.targetProjectAccountId;
+    this.targetTestingAccountId = options.targetTestingAccountId;
+    this.projectConfig = options.projectConfig;
+    this.projectDir = options.projectDir;
+    this.projectId = options.projectId;
+    this.debug = options.debug || false;
+    this.deployedBuild = options.deployedBuild;
+    this.isGithubLinked = options.isGithubLinked;
+    this.watcher = null;
+    this.uploadWarnings = {};
+    this.components = options.components;
+    this.activeApp = null;
+    this.activePublicAppData = null;
+    this.env = options.env;
+    this.publicAppActiveInstalls = null;
+    this.mostRecentUploadWarning = null;
+
+    this.projectSourceDir = path.join(
+      this.projectDir,
+      this.projectConfig.srcDir
+    );
+
+    if (
+      !this.targetProjectAccountId ||
+      !this.projectConfig ||
+      !this.projectDir
+    ) {
+      logger.log(i18n(`${i18nKey}.failedToInitialize`));
+      process.exit(EXIT_CODES.ERROR);
+    }
+  }
+
+  async setActiveApp(appUid?: string): Promise<void> {
+    if (!appUid) {
+      logger.error(
+        i18n(`${i18nKey}.missingUid`, {
+          devCommand: uiCommandReference('hs project dev'),
+        })
+      );
+      process.exit(EXIT_CODES.ERROR);
+    }
+    const app =
+      Object.values(this.components).find(
+        component => component.uid === appUid
+      ) || null;
+
+    if (app && isAppIRNode(app)) {
+      this.activeApp = app;
+
+      if (app.config.distribution === APP_DISTRIBUTION_TYPES.PUBLIC) {
+        try {
+          await this.setActivePublicAppData();
+          await this.checkActivePublicAppInstalls();
+          await this.checkPublicAppInstallation();
+        } catch (e) {
+          logError(e);
+        }
+      }
+    }
+
+    return;
+  }
+
+  async setActivePublicAppData(): Promise<void> {
+    const {
+      data: { results: portalPublicApps },
+    } = await fetchPublicAppsForPortal(this.targetProjectAccountId);
+
+    const activePublicAppData = portalPublicApps.find(
+      ({ sourceId }) => sourceId === this.activeApp?.uid
+    );
+
+    if (!activePublicAppData) {
+      return;
+    }
+
+    const {
+      data: { uniquePortalInstallCount },
+    } = await fetchPublicAppProductionInstallCounts(
+      activePublicAppData.id,
+      this.targetProjectAccountId
+    );
+
+    this.activePublicAppData = activePublicAppData;
+    this.publicAppActiveInstalls = uniquePortalInstallCount;
+  }
+
+  async checkActivePublicAppInstalls(): Promise<void> {
```

Comment:
Also not a blocker, but we may be able to set ourselves up for the future a little bit by trying to break some of the app-specific logic out into its own file(s)

Reply:
Oh that's a good call. I'll see if there's anything I can do right now. Might need to be part of a bigger revamp since some of the logic is pretty intertwined with the LocalDevManager (for example, the idea of having an `activePublicApp` that's attached to the instance)

Reply:
In the short-term we only support a single app, which might make splitting this up easier. But we can dig more into this in a follow up PR